src/parser/jobs/gnb/modules/NoMercy.tsx

```diff
@@ -37,228 +34,87 @@ const EXPECTED_USES = {
 			},
-			this.onRemoveNoMercy,
-		)
-		this.addEventHook('complete', this.onComplete)
-	}
-
-	private onCast(event: CastEvent) {
-		const actionId = event.ability.guid
 
-		if (actionId === ACTIONS.ATTACK.id) {
-			return
-		}
+			{
+				action: ACTIONS.BURST_STRIKE,
+				expectedPerWindow: EXPECTED_USES.BURST_STRIKE,
+			},
 
-		if (actionId === ACTIONS.NO_MERCY.id) {
-			const noMercyState = new NoMercyState(event.timestamp)
-			const fightTimeRemaining = this.parser.fight.end_time - event.timestamp
-			noMercyState.isRushing = NO_MERCY_BUFF_DURATION >= fightTimeRemaining
+			{
+				action: ACTIONS.SONIC_BREAK,
+				expectedPerWindow: EXPECTED_USES.SONIC_BREAK,
+			},
 
-			this.noMercyWindows.push(noMercyState)
-		}
+			{
+				action: ACTIONS.BLASTING_ZONE,
+				expectedPerWindow: EXPECTED_USES.BLASTING_ZONE,
+			},
 
-		// So long as we're in this window, log our actions to it.
-		const lastNoMercy = this.lastNoMercy
-		if (lastNoMercy != null && lastNoMercy.end == null) {
-			lastNoMercy.rotation.push(event)
+			{
+				action: ACTIONS.BOW_SHOCK,
+				expectedPerWindow: EXPECTED_USES.BOW_SHOCK,
+			},
 
-			const action = getDataBy(ACTIONS, 'id', actionId) as TODO
-			if (!action) { return }
+			{
+				action: ACTIONS.ROUGH_DIVIDE,
+				expectedPerWindow: EXPECTED_USES.ROUGH_DIVIDE,
+			},
+		],
 
-			// Pre-process on the number of certain things we did
-			if (action.onGcd) {
-				lastNoMercy.numGcds++
-			}
+		suggestionContent: <Trans id="gnb.nomercy.suggestions.expected-uses.content">
+			Watch your uses of certain abilities during <ActionLink {...ACTIONS.NO_MERCY} />. Under ideal conditions, you should
+				be using <ActionLink {...ACTIONS.SONIC_BREAK} />, a full <ActionLink {...ACTIONS.GNASHING_FANG} /> combo, and all of
+				your off-GCD skills - <ActionLink {...ACTIONS.BLASTING_ZONE} />, <ActionLink {...ACTIONS.BOW_SHOCK} />, and at least one
+				charge of <ActionLink {...ACTIONS.ROUGH_DIVIDE} /> - under the buff duration.
+		</Trans>,
+		severityTiers: SEVERITIES.MISSING_EXPECTED_USES,
 
-			switch (actionId) {
-			// Blasting Zone is just a traited version of Danger Zone
-			// Impossible to see both in the same log outside of hacking
-			case ACTIONS.BLASTING_ZONE.id:
-			case ACTIONS.DANGER_ZONE.id:
-				lastNoMercy.numBlastingZones++
-				break
-			case ACTIONS.SONIC_BREAK.id:
-				lastNoMercy.numSonicBreaks++
-				break
-			case ACTIONS.ROUGH_DIVIDE.id:
-				lastNoMercy.numRoughDivides++
-				break
-			case ACTIONS.GNASHING_FANG.id:
-				lastNoMercy.numGnashingFangs++
-				break
-			case ACTIONS.BOW_SHOCK.id:
-				lastNoMercy.numBowShocks++
-				break
-			}
-		}
 	}
 
-	private onRemoveNoMercy(event: BuffEvent) {
-		const lastNoMercy = this.lastNoMercy
+	changeExpectedTrackedActionClassLogic(buffWindow: BuffWindowState, action: BuffWindowTrackedAction) {
 
-		if (lastNoMercy != null) {
-			lastNoMercy.end = event.timestamp
-		}
-	}
+		if (action.action.id === ACTIONS.BURST_STRIKE.id) {
```

Comment:
Nit, I'd consider changing the name of the param here if possible just so it's a bit less weird to read than `action.action`.

Reply:
changed to const Id

Reply:
I'd still rename the param (and leading upper case names like this trigger me ngl), it's awkward to read either way.